### PR TITLE
Extract media controls into method

### PR DESCRIPTION
**Pull request by balloob**
## Proposed change

Extract button generation into a new method to make it easier to follow what's going on. 

Update gallery to show more relevant states.

If player is idle and it supports play, show play button.

Check for existence of "media_position" instead of truthy check, as we were not showing progress bars if media position was 0.

Updated gallery:
image

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: 

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/lovelace/cards/hui-media-control-card.ts [hunk 1]
```diff
@@ -118,7 +124,7 @@ export class HuiMediaControlCard extends LitElement implements LovelaceCard {
       return;
     }
 
-    const stateObj = this.hass.states[this._config.entity] as MediaEntity;
+    const stateObj = this._stateObj;
```
Comment (zsarnett) [inline src/panels/lovelace/cards/hui-media-control-card.ts]: Where is this defined? I feel like i'm missing it somewhere

File: src/panels/lovelace/cards/hui-media-control-card.ts [hunk 2]
```diff
@@ -130,7 +136,7 @@ export class HuiMediaControlCard extends LitElement implements LovelaceCard {
       stateObj.state === "playing"
```
Comment (zsarnett) [inline src/panels/lovelace/cards/hui-media-control-card.ts]: I can't comment on the line above for some reason but should it be 
```js
this._showProgressBar !== false
```

File: src/panels/lovelace/cards/hui-media-control-card.ts [hunk 3]
```diff
@@ -427,16 +382,81 @@ export class HuiMediaControlCard extends LitElement implements LovelaceCard {
 
     if (this._image !== oldImage) {
       this._setColors();
-      return;
     }
   }
 
+  private _getControls(): ControlButton[] | undefined {
+    const stateObj = this.hass!.states[this._config!.entity] as MediaEntity;
```
Comment (zsarnett) [inline src/panels/lovelace/cards/hui-media-control-card.ts]: Why not use `this._stateObj`

**Comment by zsarnett** [comment]
Idle is still Grayscale

File: src/panels/lovelace/cards/hui-media-control-card.ts (hunk 2 above)
Comment (balloob) [reply src/panels/lovelace/cards/hui-media-control-card.ts]: showProgressBar returns a boolean.

File: src/panels/lovelace/cards/hui-media-control-card.ts (hunk 1 above)
Comment (balloob) [reply src/panels/lovelace/cards/hui-media-control-card.ts]: line 543.

File: src/panels/lovelace/cards/hui-media-control-card.ts [hunk 4]
```diff
@@ -264,80 +273,19 @@ export class HuiMediaControlCard extends LitElement implements LovelaceCard {
...
-                                    : html`
-                                        <paper-icon-button
-                                          icon="hass:skip-next"
-                                          .action=${"media_next_track"}
-                                          @click=${this._handleClick}
-                                        ></paper-icon-button>
-                                      `}
-                                </div>
-                              `}
+                        <div class="playback-controls">
```
Comment (zsarnett) [inline src/panels/lovelace/cards/hui-media-control-card.ts]: Update the CSS class to be `.playback-controls` or change this back to `.controls`

File: src/panels/lovelace/cards/hui-media-control-card.ts (hunk 3 above)
Comment (balloob) [reply src/panels/lovelace/cards/hui-media-control-card.ts]: doh

File: src/panels/lovelace/cards/hui-media-control-card.ts [hunk 5]
```diff
@@ -398,13 +352,15 @@ export class HuiMediaControlCard extends LitElement implements LovelaceCard {
       applyThemesOnElement(this, this.hass.themes, this._config.theme);
     }
 
+    this._updateProgressBar();
+
     if (
       !this._progressInterval &&
-      this._showProgressBar &&
+      this._showProgressBar !== false &&
```
Comment (zsarnett) [inline src/panels/lovelace/cards/hui-media-control-card.ts]: Why check this to false if its a boolean

Comment (balloob) [reply src/panels/lovelace/cards/hui-media-control-card.ts]: oops old stuff.

**Comment by balloob** [comment]
Updated 
image

**Comment by balloob** [comment]
image

**Review by zsarnett — APPROVED** [review]

**Pull request merged**